src/Runner.Worker/FileCommandManager.cs

```diff
@@ -259,4 +259,73 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
             return text.Substring(originalIndex, lfIndex - originalIndex);
         }
     }
+
+    public sealed class SetStepSummaryCommand : RunnerService, IFileCommandExtension
+    {
+        private const int _attachmentSizeLimit = 128 * 1024;
+
+        public string ContextName => "step_summary";
+        public string FilePrefix => "step_summary_";
+
+        public Type ExtensionType => typeof(IFileCommandExtension);
+
+        public void ProcessCommand(IExecutionContext context, string filePath, ContainerInfo container)
+        {
+            if (!context.Global.Variables.GetBoolean("DistributedTask.UploadStepSummary") ?? true)
+            {
+                Trace.Info("Step Summary is disabled; skipping attachment upload");
+                return;
+            }
+
+            if (String.IsNullOrEmpty(filePath) || !File.Exists(filePath))
+            {
+                Trace.Info($"Step Summary file ({filePath}) does not exist; skipping attachment upload");
+                return;
+            }
+
+            var fileSize = new FileInfo(filePath).Length;
```

Comment:
`new FileInfo()` can throw on permission error with the file.

Reply:
I've moved the `try` block up by a few lines so exceptions that might be happening here are caught as well.